### PR TITLE
Add short-hand methods for text and blob types in MySQL

### PR DESCRIPTION
**Pull request by kamipo**
In Pg and Sqlite3, `:text` and `:binary` have variable unlimited length.
But in MySQL, these have limited length for each types (see also #21591, #21619).
This change adds short-hand methods for each text and blob types.

Before:

``` ruby
create_table :foos do |t|
  t.blob   :tiny_blob,   limit: 255
  t.binary :medium_blob, limit: 16777215
  t.binary :long_blob,   limit: 2147483647
  t.text   :tiny_text,   limit: 255
  t.text   :medium_text, limit: 16777215
  t.text   :long_text,   limit: 2147483647
end
```

Short-hand methods:

``` ruby
create_table :foos do |t|
  t.tinyblob   :tiny_blob
  t.mediumblob :medium_blob
  t.longblob   :long_blob
  t.tinytext   :tiny_text
  t.mediumtext :medium_text
  t.longtext   :long_text
end
```

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb [hunk 1]
```diff
@@ -1163,8 +1187,9 @@ def create_table_definition(name, temporary = false, options = nil, as = nil) #
       def binary_to_sql(limit) # :nodoc:
         case limit
         when 0..0xfff;           "varbinary(#{limit})"
-        when nil;                "blob"
-        when 0x1000..0xffffffff; "blob(#{limit})"
+        when nil, 0x1000..0xffff;   'blob'
+        when 0x10000..0xffffff;     'mediumblob'
+        when 0x1000000..0xffffffff; 'longblob'
```
Comment (matthewd) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: Why this change? It seems better to leave this up to MySQL to me.

Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: This is the same way as [text_to_sql](https://github.com/kamipo/rails/blob/add_text_and_blob_shorthand_methods/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L1209-L1217):

```ruby
      def text_to_sql(limit) # :nodoc:
        case limit
        when 0..0xff;               'tinytext'
        when nil, 0x100..0xffff;    'text'
        when 0x10000..0xffffff;     'mediumtext'
        when 0x1000000..0xffffffff; 'longtext'
        else raise(ActiveRecordError, "No text type has byte length #{limit}")
        end
      end
```

`blob(2147483647)` is not usual. It is `longblob` usually.

**Comment by jeremy** [comment]
I like this. We've seen a lot of stubbed toes from people trying to figure out the correct `:size` to trigger the desired MySQL type.

**Comment by pixeltrix** [comment]
Rather than using `t.mediumblob :column` why not an option, e.g: `t.binary :column, size: :medium` - that way the PostgreSQL adapter can just ignore it and the migration is still database independent. I know that trying to write database independent schema.rb is a losing battle so it probably doesn't make a difference in the long run. :smile:

**Comment by jeremy** [comment]
The size label is an improvement too. People look for known terminology, though: "where is bigint" and "I need a mediumtext."
